src/appleseed-max-impl/maxtextures/maxproctexturesource.cpp

```diff
@@ -0,0 +1,100 @@
+
+// Interface
+#include "maxproctexturesource.h"
+
+// appleseed.foundation headers.
+#include "foundation/utility/siphash.h"
+
+// 3ds max include
+#include "imtl.h"
+#include "maxapi.h"
+
+MaxProceduralSource::MaxProceduralSource(
+    Texmap* max_texmap)
+    : m_texmap(max_texmap)
+{
+}
+
+MaxProceduralSource::~MaxProceduralSource()
+{
+}
+
+foundation::uint64 MaxProceduralSource::compute_signature() const
+{
+    return foundation::siphash24(m_texmap, sizeof(*m_texmap));
+}
+
+//--- Shade Context
+class MaxProceduralSource::MaxSC
```

Comment:
Wrap in anonymous namespace, and don't make the class a child of `MaxProceduralSource`.